app/models/option_set.rb

```diff
@@ -293,6 +293,23 @@ def as_json(options = {})
     end
   end
 
+  def worksheet_name
+    name = self.name
+    name = name.truncate(31) if self.name.size > 31
+    name = name.gsub(
+      /[\[\]\*\\?\:\/]/, {
+        '[' => '(',
+        ']' => ')',
+        '*' => '∗',
+        '?' => '',
+        ':' => '-',
+        '\\' => '-',
+        '/' => '-'
+      }
+    )
+    name
+  end
+
```

Comment:
I probably could find _somewhere_ to define it where that particular kind of view has access to it, I can try to relocate this piece, but the first couple of places I tried didn't work.